exporter/collector/logs.go

```diff
@@ -527,8 +501,54 @@ func (l logMapper) logToSplitEntries(
 		return entries, nil
 	}
 
-	entry.Payload = payload
-	return []logging.Entry{entry}, nil
+	switch p := payload.(type) {
+	case string:
+		entry.Payload = &logpb.LogEntry_TextPayload{TextPayload: p}
+	case *anypb.Any:
+		entry.Payload = &logpb.LogEntry_ProtoPayload{ProtoPayload: p}
+	default:
+		jb, err := json.Marshal(p)
+		if err != nil {
+			return nil, fmt.Errorf("logging: json.Marshal: %w", err)
+		}
+		var verify map[string]interface{}
+		if err := json.Unmarshal(jb, &verify); err == nil {
+			entry.Payload = &logpb.LogEntry_JsonPayload{JsonPayload: jsonMapToProtoStruct(verify)}
+		}
+	}
+
+	return []logpb.LogEntry{entry}, nil
+}
+
+func jsonMapToProtoStruct(m map[string]interface{}) *structpb.Struct {
```

Comment:
this is directly copied from how the logging client converts with `ToLogEntry`: https://github.com/googleapis/google-cloud-go/blob/69705144832c715cf23832602ad9338b911dff9a/logging/logging.go#L853